src/s3_fetchers.rs

```diff
@@ -129,25 +149,9 @@ async fn fetch_shard_or_retry(
                     }
                 };
 
-                let indexer_shard = match serde_json::from_slice::<
+                break Ok(serde_json::from_slice::<
                     near_indexer_primitives::IndexerShard,
-                >(body_bytes.as_ref())
-                {
-                    Ok(indexer_shard) => indexer_shard,
-                    Err(err) => {
-                        tracing::debug!(
-                            target: crate::LAKE_FRAMEWORK,
-                            "Failed to parse the {:0>12}/shard_{}.json. Retrying in 1s...\n {:#?}",
-                            block_height,
-                            shard_id,
-                            err,
-                        );
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        continue;
-                    }
-                };
-
-                break indexer_shard;
+                >(body_bytes.as_ref())?);
```

Comment:
I would follow the same pattern as for `block_view` in `fetch_streamer_message` function, where you `break bytes_stream.into_bytes()` and then do the parsing outside of the loop, as currently, it is hard to read this code since the return value happens in the middle of the body of the function.